### PR TITLE
leader.Become() failed when kube-api couldn't be reached during startup

### PR DESCRIPTION
**Pull request by yu2003w**
Fixed:  #87

Added poll logic in case kube-api couldn't be reached when leader.Become() is called.

Signed-off-by: Wei Yu <yuweiw@cn.ibm.com>

### CODE REVIEW

**Comment by markusthoemmes** [comment]
I'd say we just ditch leader election for now. It's not something we use in Knative yet, it'll likely not be needed here as well.

File: cmd/manager/main.go [hunk 1]
```diff
@@ -91,10 +94,15 @@ func main() {
 
 	ctx := context.TODO()
 
+	// We sometimes startup faster than we can reach kube-api. Poll on failure to prevent us terminating
 	// Become the leader before proceeding
-	err = leader.Become(ctx, "knative-serving-operator-lock")
-	if err != nil {
-		log.Error(err, "")
+	if perr := wait.PollImmediate(time.Second, 60*time.Second, func() (bool, error) {
```
Comment (houshengbo) [inline cmd/manager/main.go]: @yu2003w I do not think we will need the leader selection any longer, coz the new controller model does not need it.

Comment (yu2003w) [reply cmd/manager/main.go]: ok, I will close this. Thanks.

**Comment by yu2003w** [comment]
Close this PR as it's not applicable for current implementation.

**Pull request closed without merging**